docs/architecture.md

```diff
@@ -15,139 +15,207 @@ Prometheus instances scrape samples from various targets and then push them to C
 
 Cortex requires that each HTTP request bear a header specifying a tenant ID for the request. Request authentication and authorization are handled by an external reverse proxy.
 
-Incoming samples (writes from Prometheus) are handled by the [distributor](#distributor) while incoming reads (PromQL queries) are handled by the [query frontend](#query-frontend).
+Incoming samples (writes from Prometheus) are handled by the [distributor](#distributor) while incoming reads (PromQL queries) are handled by the [querier](#querier) or optionally by the [query frontend](#query-frontend).
+
+## Storage
+
+Cortex currently supports two storage engines to store and query the time series:
+
+- Chunks (default, stable)
+- Blocks (experimental)
+
+The two engines mostly share the same Cortex architecture with few differences outlined in the rest of the document.
+
+### Chunks storage (default)
+
+The chunks storage stores each single time series into a separate object called chunk. Each chunk contains the timestamp-value pairs for a given period (defaults to 12 hours) of the series. Series chunks are then indexed by time range period and metric labels, in order to provide a fast lookup across millions - or even billions - of chunks.
+
+For this reason, the chunks storage consists of:
+
+* An index for the chunks. This index can be backed by:
+    * [Amazon DynamoDB](https://aws.amazon.com/dynamodb)
+    * [Google Bigtable](https://cloud.google.com/bigtable)
+    * [Apache Cassandra](https://cassandra.apache.org)
+* An object store for the chunk data itself, which can be:
+    * [Amazon DynamoDB](https://aws.amazon.com/dynamodb)
+    * [Google Bigtable](https://cloud.google.com/bigtable)
+    * [Apache Cassandra](https://cassandra.apache.org)
+    * [Amazon S3](https://aws.amazon.com/s3)
+    * [Google Cloud Storage](https://cloud.google.com/storage/)
+
+Internally, the access to the chunks storage relies on a unified interface called "chunks store". Unlike other Cortex components, the chunk store is not a separate service, but rather a library embedded in the services that need to access the long-term storage: [ingester](#ingester), [querier](#querier) and [ruler](#ruler).
+
+The chunk and index data format is versioned, in order to allow Cortex operators to upgrade the cluster to the latest stable version and take advantage of new features and improvements - which may introduce changes in the storage format - without requiring any downtime or complex procedures to rewrite the stored data.
+
+A set of storage schemas are used to map the version to use to read and write time series belonging to a specific period of time. The current schema recommendation is the **v10 schema** (v11 is still experimental). For more information about the schema, please check out the "[Schema](guides/running.md#schema)" documentation.
```

Comment:
```suggestion
A set of schemas are used to map the version while reading and writing time series belonging to a specific period of time. 

The current stable schema is the **v10 schema** (v11 is still experimental). For more information about the schema, please check out the "[Schema](guides/running.md#schema)" documentation.
```

Reply:
How do you feel about this split? Furthermore, I think the first paragraph here needs to be incorporated into the previous one. In the previous paragraph, we talk about the benefits of the schema but then on this paragraph we define how the schema is used if my understanding is correct which feels counterintuitive.

I think we could do something like:

> The chunk and index format are versioned, this allows Cortex operators to upgrade the cluster to take advantage of new features and improvements. This strategy enables changes in the storage format without requiring any downtime or complex procedures to rewrite the stored data. The set of schemas are used to map the version while reading and writing time series belonging to a specific period of time.

> The current stable schema is the **v10 schema** (v11 is still experimental). For more information about the schema, please refer to the [Schema documentation](guides/running.md#schema).

Reply:
The grouping you propose makes sense to me. I picked your updated sentence and re-grouped it in the doc.